homeassistant/components/roborock/coordinator.py

```
@@ -177,222 +110,145 @@ def dock_device_info(self) -> DeviceInfo:
         """
-        dock_type = self.roborock_device_info.props.status.dock_type
+        dock_type = self.properties_api.status.dock_type
         return DeviceInfo(
-            name=f"{self.roborock_device_info.device.name} Dock",
+            name=f"{self._device.device_info.name} Dock",
             identifiers={(DOMAIN, f"{self.duid}_dock")},
             manufacturer="Roborock",
-            model=f"{self.roborock_device_info.product.model} Dock",
+            model=f"{self._device.product.model} Dock",
             model_id=str(dock_type.value) if dock_type is not None else "Unknown",
-            sw_version=self.roborock_device_info.device.fv,
+            sw_version=self._device.device_info.fv,
         )
 
-    def parse_map_data_v1(
-        self, map_bytes: bytes
-    ) -> tuple[bytes | None, MapData | None]:
-        """Parse map_bytes and return MapData and the image."""
-        try:
-            parsed_map = self.map_parser.parse(map_bytes)
-        except (IndexError, ValueError) as err:
-            _LOGGER.debug("Exception when parsing map contents: %s", err)
-            return None, None
-        if parsed_map.image is None:
-            return None, None
-        img_byte_arr = io.BytesIO()
-        parsed_map.image.data.save(img_byte_arr, format=MAP_FILE_FORMAT)
-        return img_byte_arr.getvalue(), parsed_map
-
     async def _async_setup(self) -> None:
         """Set up the coordinator."""
-        # Verify we can communicate locally - if we can't, switch to cloud api
         await self._verify_api()
-        self.api.is_available = True
+        await self.properties_api.status.refresh()
 
+        self._last_home_update_attempt = dt_util.utcnow()
         try:
-            maps = await self.api.get_multi_maps_list()
+            await self.properties_api.home.discover_home()
+        except RoborockDeviceBusy:
+            _LOGGER.info("Home discovery skipped while device is busy/cleaning")
```

what are the implications of us not doing this on setup and silently failing?